mitodl/mitxpro
# Fix rough edges with register flow

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #119 
Fixes #117 

#### What's this PR do?
This fixes two things:

- Adds validation for the password field length
- If an register confirmation code is expired or used more than once, the user gets a message

#### How should this be manually tested?
- Register and enter a password less than 8 chars, you should see a message. It should let you submit if you enter a longer one.
- Click the confirmation link a second time and you should get a message and a link to register again.

#### Screenshots (if appropriate)

Invalid confirmation code link messaging (same language as discussions):
![Screenshot_2019-04-08 mitxpro(2)](https://user-images.githubusercontent.com/28598/55749558-38796580-5a0f-11e9-8319-7446bd1a3c31.png)

Validation for password:
![Screenshot_2019-04-08 mitxpro](https://user-images.githubusercontent.com/28598/55749559-38796580-5a0f-11e9-9848-6b960866aad1.png)

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/mJHpiG1Ll82UP0HldA/giphy.gif)
